src/Middleware/RequestThrottling/perf/Microbenchmarks/QueueFullOverhead.cs

```diff
@@ -0,0 +1,79 @@
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using BenchmarkDotNet.Attributes;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Logging.Abstractions;
+using Microsoft.Extensions.Options;
+
+namespace Microsoft.AspNetCore.RequestThrottling.Microbenchmarks
+{
+    public class QueueFullOverhead
+    {
+        private const int _numRequests = 2000;
+        private int _requestCount = 0;
+        private TaskCompletionSource<bool> tcs;
+
+        private RequestThrottlingMiddleware _middleware;
+
+        [Params(1,8,64)]
+        public int MaxConcurrentRequests;
```

Comment:
This variation applies to all scenarios. Only put scenarios that vary on MaxConcurrentRequests in this class.

Reply:
If this is flaky with variations, remove the variations for now.

Reply:
reduced to only one param, this also helps with flaky tests